### ISSUE TITLE
[BUG] <Search function for convo/conversations does not work correctly>

### ISSUE THREAD

**Issue by featherridge** [oxen-io/session-ios#930]
### Code of conduct

- [X] I have read and agree to adhere to the [Code of Conduct](https://github.com/oxen-io/session-ios/blob/master/CODE_OF_CONDUCT.md)

### Self-training on how to write a bug report

- [X] I have learned [how to write a bug report](https://www.browserstack.com/guide/how-to-write-a-bug-report)

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Searching in "Contacts & Groups" does not work correctly.
Example, I have a contacted named "X"
I type "X" in, "X" does not show.

### Expected Behavior

I have a contact named "X"
I type "X" in, and "X" does show.

### Steps To Reproduce

1. In search function
2. Type a contact or keyword
3. See not relevant results show

### iOS Version

iOS 16.7

### Session Version

2.4.3 (427)

**Comment by KeeJef** [comment]
Yeah this is a bit of an issue, we only support search for strings longer than 2 characters, otherwise things get tricky performance wise. A comprise might be to allow search for usernames / group names for less than 2 characters but still limit conversation searches. @mpretty-cyro thoughts?

**Comment by mpretty-cyro** [comment]
I do wonder if users will get confused as to why they aren't getting message search results when only entering a single character

I'll take a quick look so see what the performance is like for a single character search - I think the last time I actually checked were were using FTS4 (now have FTS5) and I'm not sure whether we had a limit on the search results (it's now 500 for contacts and 500 for messages)

**Comment by KeeJef** [comment]
@mpretty-cyro sounds reasonable, lets put this in the backlog for iOS

**Comment by mpretty-cyro** [comment]
Jira ticket is SES-1321
